app/com/arpnetworking/metrics/portal/alerts/impl/DatabaseAlertRepository.java

```diff
@@ -335,29 +277,12 @@ private NagiosExtension convertToEbeanNagiosExtension(@Nullable final models.int
         public PagedList<models.ebean.Alert> createAlertQuery(final AlertQuery query) {
             ExpressionList<models.ebean.Alert> ebeanExpressionList = Ebean.find(models.ebean.Alert.class).where();
             ebeanExpressionList = ebeanExpressionList.eq("organization.uuid", query.getOrganization().getId());
-            if (query.getCluster().isPresent()) {
-                ebeanExpressionList = ebeanExpressionList.eq("cluster", query.getCluster().get());
-            }
-            if (query.getContext().isPresent()) {
-                ebeanExpressionList = ebeanExpressionList.eq("context", query.getContext().get().toString());
-            }
-            if (query.getService().isPresent()) {
-                ebeanExpressionList = ebeanExpressionList.eq("service", query.getService().get());
-            }
 
             //TODO(deepika): Add full text search [ISSUE-11]
             if (query.getContains().isPresent()) {
                 final Junction<models.ebean.Alert> junction = ebeanExpressionList.disjunction();
                 ebeanExpressionList = junction.contains("name", query.getContains().get());
-                if (!query.getCluster().isPresent()) {
-                    ebeanExpressionList = junction.contains("cluster", query.getContains().get());
-                }
-                if (!query.getService().isPresent()) {
-                    ebeanExpressionList = junction.contains("service", query.getContains().get());
-                }
-                ebeanExpressionList = junction.contains("metric", query.getContains().get());
-                ebeanExpressionList = junction.contains("statistic", query.getContains().get());
-                ebeanExpressionList = junction.contains("operator", query.getContains().get());
+                ebeanExpressionList = junction.contains("query", query.getContains().get());
```

Comment:
Is this case insensitive? How does it handle locale?

Reply:
So I think we add the full-text index here, and file an issue for the locale which is fucked everywhere.

Reply:
https://aws.amazon.com/blogs/database/performing-parallel-queries-and-phrase-searching-with-amazon-rds-for-postgresql-9-6-1/